### ISSUE TITLE
feat: blockful custom erc20 as the mock contract

### ISSUE THREAD

**Issue by 0xneves** [blockful-io/swaplace-contracts#92]
Prosposal: 
- get the blockful erc20 from [this repo](https://github.com/blockful-io/blockful-contracts/blob/main/contracts/token/ERC20/ERC20.sol)
- use the blockful erc20 instead of the openzeppelin in the swaplace repo
- test for bugs and conflicts and propose the fixes as well (if needed)

**Comment by 0xneves** [comment]
Well go for it, can't barely wait to see the PR @RafaDSan 👍

**Comment by 0xneves** [comment]
Hey @RafaDSan , how is the task going? Do you need anything to help you with?

**Comment by 0xjoaovpsantos** [comment]
Hey @0xneves, if this task isn't started, I'm interested to get

**Comment by RafaDSan** [comment]
@0xneves I've added some new tests to the `TestMockContracts.test.ts` file but I am having some problems importing the ERC20 code. 
Here is what I am doing:

1. Copied the erc20 contract of  blockful and pasted in the MockERC20.sol
2. Changed the line `abstract contract ERC20 is IERC20, IERC20Permit, IERC20Errors` for and removed the `abstract`
3, Added the following files to the `interfaces` directory:  `IERC20.SOL`, `IERC20Errors.sol`, `IERC20Permit.sol`
4. Run the command `npx hardhat test`  and receive the following error `Error: missing argument:  in Contract constructor (count=0, expectedCount=2, code=MISSING_ARGUMENT, version=contracts/5.7.0)` which I am trying to figure it out rn.
